sensor_test.py

```diff
@@ -0,0 +1,184 @@
+    sensor_info = get_sensor_info()
+
+    sensor_name = sensor_info.get("name")
+    assert isinstance(sensor_name, str), "Sensor name is not a string"
+
+    sensor_hid = sensor_info.get("hid")
+    assert isinstance(sensor_hid, str), "Sensor hid is not a string"
+
+    sensor_model = sensor_info.get("model")
+    assert isinstance(sensor_model, str), "Sensor model is not a string"
+
+    sensor_firmware_version = sensor_info.get("firmware_version")
+    assert isinstance(
+        sensor_firmware_version, int
+    ), "Sensor firmware version is not a int"
+
+    sensor_reading_interval = sensor_info.get("reading_interval")
+    assert isinstance(
+        sensor_reading_interval, int
+    ), "Sensor reading interval is not a string"
+
+    sensor_reading = get_sensor_reading()
+    assert isinstance(
+        sensor_reading, float
+    ), "Sensor doesn't seem to register temperature"
+
+    print("Sanity test passed")
+
+
+def test_reboot(get_sensor_info, reboot_sensor):
+    """
+    Steps:
+        1. Get original sensor info.
+        2. Reboot sensor.
+        3. Wait for sensor to come back online.
+        4. Get current sensor info.
+        5. Validate that info from Step 1 is equal to info from Step 4.
+    """
+    print("Get original sensor info")
+    sensor_info_before_reboot = get_sensor_info()
+
+    print("Reboot sensor")
+    reboot_response = reboot_sensor()
+    assert reboot_response == "rebooting", "Sensor did not return proper text in response to reboot request"
+
+    print("Wait for sensor to come back online")
+    sensor_info_after_reboot = wait(
+        func=get_sensor_info,
+        condition=lambda x: isinstance(x, dict),
+        tries=10,
+        timeout=1,
+    )
+
+    print("Validate that info from Step 1 is equal to info from Step 4")
+    assert sensor_info_before_reboot == sensor_info_after_reboot, "Sensor info after reboot doesn't match sensor info before reboot"
+
+
+def test_set_sensor_name(get_sensor_info, set_sensor_name):
+    """
+    1. Set sensor name to "new_name".
+    2. Get sensor_info.
+    3. Validate that current sensor name matches the name set in Step 1.
+    """
+    # Step 1: Set sensor name to "new_name"
+    set_sensor_name("new_name")
+    
+    # Step 2: Get sensor_info
+    sensor_info = get_sensor_info()
+    
+    # Step 3: Validate that current sensor name matches the name set in Step 1
+    assert sensor_info['name'] == "new_name", f"Sensor name should be 'new_name', but got '{sensor_info['name']}'"
+
+
+def test_set_sensor_reading_interval(
+    get_sensor_info, set_sensor_reading_interval, get_sensor_reading
+):
+    """
+    1. Set sensor reading interval to 1.
+    2. Get sensor info.
+    3. Validate that sensor reading interval is set to interval from Step 1.
+    4. Get sensor reading.
+    5. Wait for interval specified in Step 1.
+    6. Get sensor reading.
+    7. Validate that reading from Step 4 doesn't equal reading from Step 6.
+    """
```

Comment:
You can remove `"""` here and on line 91 and have all the documentation in a single block.